src/lib.rs

```diff
@@ -585,6 +585,57 @@ impl<T: Debug + PrimInt> Vob<T> {
         }
     }
 
+    /// Counts the number of set bits.
+    /// This method assumes the range is processed with process_range()
+    fn count_set_bits(&self, range: Range<usize>) -> usize {
+        // Early return for empty ranges
+        if range.start >= range.end {
```

Comment:
Is such a range valid? I suspect we could, and probably should, `panic` in such a case. It would be good to know what the std library tends to do in such cases.

Reply:
I was certain it wasn't valid, but turns out it is just an alias for `range.is_empty()`, and `std` seems to happily construct one

https://doc.rust-lang.org/src/core/ops/range.rs.html#147-149
```
    pub fn is_empty(&self) -> bool {

        !(self.start < self.end)

    }
```
https://doc.rust-lang.org/std/ops/struct.Range.html#method.is_empty
> Returns true if the range contains no items.

Reply:
if nothing else, it's a catch for bad user input. iter_set_bits(234234..25232344) will make process_range() return (len(),len()) (if the ranges are out of bounds)
If we don't want to allow bad input like that, we'd need to update process_range() with a Result return type